### PR TITLE
Extension: Add HexFloatLiterals

### PR DESCRIPTION
**Pull request by bgamari**
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [x] Any changes that could be relevant to users have been recorded in the changelog.
* [x] The documentation has been updated, if necessary.
* [x] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

No testing performed.

### CODE REVIEW

File: Cabal/changelog [hunk 1]
```diff
@@ -18,12 +18,9 @@
 	* Added '.Lens' modules, with optics for package description data
 	  types. (#4701)
 	* Support for GHC's numeric -g debug levels (#4673).
-	* Added elif-conditionals to .cabal syntax (#4750).
```
Comment (23Skidoo) [inline Cabal/changelog]: Looks like a botched merge. Otherwise LGTM.

**Comment by 23Skidoo** [comment]
Please fix the changelog changes, feel free to merge after that.

**Comment by angerman** [comment]
How can I take over this PR? I do have some free time and could just fix the merge conflict and this could go off our radars?

**Comment by 23Skidoo** [comment]
@angerman Sure, go ahead. Open a new PR if you can't push changes to @bgamari's branch.

**Comment by 23Skidoo** [comment]
Closing in favour of #4885.

**Pull request closed without merging**